services/API-service/src/api/country/country.service.spec.ts

```diff
@@ -0,0 +1,293 @@
+          useClass: Repository,
+        },
+      ],
+    }).compile();
+
+    service = module.get<CountryService>(CountryService);
+    countryRepository = module.get<Repository<CountryEntity>>(
+      getRepositoryToken(CountryEntity),
+    );
+    disasterRepository = module.get<Repository<DisasterEntity>>(
+      getRepositoryToken(DisasterEntity),
+    );
+    countryDisasterSettingsRepository = module.get<
+      Repository<CountryDisasterSettingsEntity>
+    >(getRepositoryToken(CountryDisasterSettingsEntity));
+    leadTimeRepository = module.get<Repository<LeadTimeEntity>>(
+      getRepositoryToken(LeadTimeEntity),
+    );
+    notificationInfoRepository = module.get<Repository<NotificationInfoEntity>>(
+      getRepositoryToken(NotificationInfoEntity),
+    );
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('getCountries', () => {
+    it('should return an array of countries', async () => {
+      const result = [new CountryEntity()];
+      jest.spyOn(countryRepository, 'find').mockResolvedValue(result);
+
+      expect(await service.getCountries()).toBe(result);
+      expect(countryRepository.find).toHaveBeenCalledWith({ relations });
+    });
+
+    it('should return an array of countries with specific country codes', async () => {
+      const countryCodes = countries
+        .map((country) => country.countryCodeISO3)
+        .join(',');
+      const result = [new CountryEntity()];
+      jest.spyOn(countryRepository, 'find').mockResolvedValue(result);
+
+      expect(await service.getCountries(countryCodes)).toBe(result);
+      expect(countryRepository.find).toHaveBeenCalledWith({
+        where: { countryCodeISO3: In(countryCodes.split(',')) },
+        relations,
+      });
+    });
+  });
+
+  describe('findOne', () => {
+    it('should return a single country with default relations', async () => {
+      const countryCodeISO3 = countries[0].countryCodeISO3;
+      const result = new CountryEntity();
+      jest.spyOn(countryRepository, 'findOne').mockResolvedValue(result);
+
+      expect(await service.findOne(countryCodeISO3)).toBe(result);
+      expect(countryRepository.findOne).toHaveBeenCalledWith({
+        where: { countryCodeISO3 },
+        relations,
+      });
+    });
+
+    it('should return a single country with custom relations', async () => {
+      const countryCodeISO3 = countries[0].countryCodeISO3;
+      const customRelations = relations.slice(0, 2);
+      const result = new CountryEntity();
+      jest.spyOn(countryRepository, 'findOne').mockResolvedValue(result);
+
+      expect(await service.findOne(countryCodeISO3, customRelations)).toBe(
+        result,
+      );
+      expect(countryRepository.findOne).toHaveBeenCalledWith({
+        where: { countryCodeISO3 },
+        relations: customRelations,
+      });
+    });
+  });
+
+  describe('addOrUpdateCountries', () => {
+    it('should add or update countries', async () => {
```

Comment:
this is quite a long test. I think it helps for readability already to add the //Arrange, //Act and //Assert steps in the code.

Reply:
I further refactored the test. Split into //Arrange, //Act and //Assert.